spring-cloud-starter-stream-processor-tensorflow/src/main/java/org/springframework/cloud/stream/app/tensorflow/processor/TensorflowProcessorProperties.java

```diff
@@ -46,6 +46,22 @@
 	 */
 	private int outputIndex = 0;
 
+	/**
+	 * Controls if the inference result is stored in the {@link org.springframework.messaging.Message} payload
+	 * or header.
+	 * By default (false) the result is stored in the payload. When set to true, the result will be
+	 * stored in the header using the resultHeaderName value.
+	 */
+	private boolean saveResultInHeader = false;
+
+	/**
+	 * Applicable only for `saveResultInHeader=true`, this property holds the header name used to store
+	 * the inference result. The default name is `tf_result`.
+	 * If multiple tensorflow processors are connected in a single pipeline then make sure they have
+	 * distinct resultHeaderName values to avoid overriding.
+	 */
+	private String resultHeaderName = "tf_result";
```

Comment:
I'd say just one `resultHeader` (even without `Name`) would kill both options.
More over, let's think if really need a tensor result in the `payload` at all.
Looks like in most cases it calculates some aspect of the `payload` and that really looks organic to enrich that info into headers.

I suggest to always have a header and by default like `tensorflow_result`. Imaging we send the data to downstream. The people should be able to determine what the header for.

If end-user configure like `result-header= ` explicitly we really store result directly into a new `payload`.

Just my thoughts on the matter.

Feel free to reject such a suggestion and share your mind as well!